### PR TITLE
Assets documentation

### PR DESCRIPTION
**Pull request by benjaminwil**
I’ve rewritten and expanded the documentation about assets in Solidus. I also split it into two articles.

The changes aren't incredibly significant, but here's a summary:

- I link back to relevant Rails documentation about manifests and the standard asset pipeline.
- When referencing the Solidus source in examples, I've made sure to update it to code that actually makes sense in master. (Previous references wouldn't have made sense if you looked at the corresponding source now.)
- I've highlighted how we recommend scoping assets to `frontend` and `backend` so nothing conflicts and how we *don't* really recommend overriding in the long term.
- I've stated clearly that the articles assume a typical Solidus installation (with `solidus_frontend` and `solidus_backend`).

In the future, I'd like to see more information about Solidus's asset manifests added. 

_This is part a larger project to improve Solidus's documentation. [See this gist with the high-level table of contents](https://gist.github.com/benjaminwil/5e2d2da3d9def6b949b833b1aa2549d3). Where and how this documentation will exist is still up for discussion._

### CODE REVIEW

File: guides/assets/asset-management.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+For more information about the asset pipeline, see the sections below. Here's a
+point-form summary of how you can get started with assets:
+
+- External JavaScript libraries, stylesheets, and image should be located in the
+  `vendor/assets` directory. Otherwise, add custom assets to your project's
+  `app/assets` tree.
+- Manifests (the `all.css`, `all.js`, and `application.js` files in your
+  project's `assets` trees) requires your app's external libraries or custom
+  assets – including any any files or directories you add deeper in the
```
Comment (tvdeyen) [inline guides/assets/asset-management.md]: ~any~ any

File: guides/assets/asset-management.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+`images`, `javascripts`, or `stylesheets`.
+
+Solidus takes this further by scoping each asset type by `spree/frontend` or
+`spree/backend`, depending on where the asset is being used.
+
+The structure for the `app` and `vendor` trees looks like this:
+
+```
+app|vendor
+|-- assets
```
Comment (tvdeyen) [inline guides/assets/asset-management.md]: I know this very picky, but could we use box drawing characters here? 😄 

```
app|vendor
└─ assets
   ├─ images
   │  └─ spree
   │     ├─ frontend
   │     └─ backend
   ├─ javascripts
   │  └─ spree
   │     ├─ frontend
   │     └─ backend
   └─ stylesheets
      └─ spree
         ├─ frontend
         └─ backend
```

File: guides/assets/asset-management.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    |       |   |-- all.js
+    |       |-- backend
+    |           |-- all.js
+    |-- stylesheets
+        |-- spree
+            |-- frontend
+            |   |-- all.css
+            |-- backend
+                |-- all.css
+```
```
Comment (tvdeyen) [inline guides/assets/asset-management.md]:
```
vendor
└─ assets
   ├─ javascripts
   │  └─ spree
   │     ├─ frontend
   │     │  └─ all.js
   │     └─ backend
   │        └─ all.js
   └─ stylesheets
      └─ spree
         ├─ frontend
         │  └─ all.css
         └─ backend
            └─ all.css
```

I ❤️ box drawing characters

File: guides/assets/asset-management.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+-->
+
+## Managing application assets
+
+We recommend using the [Rails's suggested approach to asset
+organization][asset-organization], then scoping your custom files to the
+`spree/frontend` or `spree/backend` subdirectories to avoid conflicts or
+accidental file overrides.
+
+For example, if you want to use the [Foundation CSS framework][foundation] in
```
Comment (tvdeyen) [inline guides/assets/asset-management.md]: Missing link

File: guides/assets/asset-management.md (hunk 3 above)
Comment (benjaminwil) [reply guides/assets/asset-management.md]: ❤️ Agreed

File: guides/assets/asset-management.md (hunk 4 above)
Comment (benjaminwil) [reply guides/assets/asset-management.md]: I added this at the bottom of the section. Thank you.

**Comment by benjaminwil** [comment]
@tvdeyen Hey, I updated the asset management guide per your review. When you have a chance could you re-review?

**Pull request merged**